### PR TITLE
Migrate samples for KeyVault

### PR DESCRIPTION
**Pull request by dvbb**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/keyvault/Directory.Build.props [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup Condition="$(MSBuildProjectName.EndsWith('.Samples'))">
-    <IsTestProject>true</IsTestProject>
+    <IsSampleProject>true</IsSampleProject>
```
Comment (heaths) [inline sdk/keyvault/Directory.Build.props:3]: Do not change this. It breaks all the other samples under sdk/keyvault.

**Review by heaths — DISMISSED** [review]

**Review by Yao725 — COMMENTED** [review]
Please resolve the comment [here](https://github.com/Azure/azure-sdk-for-net/pull/33224/files#r1059700408), thanks.

File: sdk/keyvault/Directory.Build.props (hunk 1 above)
Comment (dvbb) [reply sdk/keyvault/Directory.Build.props]: Hi, @heaths I pushed a commit to avoid that config impacting MGMT SDK, could you review it again thanks.

**Review by ArthurMa1978 — APPROVED** [review]

File: sdk/keyvault/Azure.ResourceManager.KeyVault/Directory.Build.props [hunk 2]
```diff
@@ -1,4 +1,8 @@
 ﻿<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup Condition="$(MSBuildProjectName.EndsWith('.Samples'))">
+    <IsSampleProject>true</IsSampleProject>
```
Comment (Yao725) [inline sdk/keyvault/Azure.ResourceManager.KeyVault/Directory.Build.props:3]: Do we need to add the property `IsSampleProject` here? I don't find it affecting anything.

Comment (dvbb) [reply sdk/keyvault/Azure.ResourceManager.KeyVault/Directory.Build.props]: removed

File: sdk/keyvault/Directory.Build.props [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <PropertyGroup Condition="$(MSBuildProjectName.EndsWith('.Samples'))">
-    <IsSampleProject>true</IsSampleProject>
+  <PropertyGroup Condition="$(MSBuildProjectName.EndsWith('.Samples'))And!$(MSBuildProjectName.StartsWith('Azure.ResourceManager'))">
```
Comment (heaths) [inline sdk/keyvault/Directory.Build.props:2]: Not sure this even works, but please add spaces for readability/maintainability.
```suggestion
  <PropertyGroup Condition="$(MSBuildProjectName.EndsWith('.Samples')) And !$(MSBuildProjectName.StartsWith('Azure.ResourceManager'))">
```

**Review by heaths — CHANGES_REQUESTED** [review]

**Review by heaths — APPROVED** [review]

**Pull request merged**